kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/MobiComConversationFragment.java

```diff
@@ -4563,10 +4567,33 @@ private void checkForCustomInput() {
 
     private void toggleAttachmentLayoutVisibility() {
         if (alCustomizationSettings.getHideAttachmentOptionsWithBots() && attachmentIconLayout != null) {
-            Contact assigneeContact = appContactService.getContactById(channel.getConversationAssignee());
-            boolean isBotAssignee = User.RoleType.BOT.getValue().equals(assigneeContact.getRoleType());
+            attachmentIconLayout.setVisibility(
+                    getCurrentConversationAssignee() == User.RoleType.BOT
+                            ? GONE : VISIBLE
+            );
+        }
+    }
+
+    private void toggleHideChatBarWithBots() {
+        if (!alCustomizationSettings.getHideChatBarWithBots() || kmMessageLinearLayout == null) {
+            return;
+        }
+        kmMessageLinearLayout.setVisibility(
+                getCurrentConversationAssignee() == User.RoleType.BOT ? GONE : VISIBLE
+        );
+    }
 
-            attachmentIconLayout.setVisibility(isBotAssignee ? GONE : VISIBLE);
+    private User.RoleType getCurrentConversationAssignee() {
+        Contact assigneeContact = appContactService.getContactById(channel.getConversationAssignee());
+        switch (assigneeContact.getRoleType()) {
+            case 1 : return User.RoleType.BOT;
+            case 2 : return User.RoleType.APPLICATION_ADMIN;
+            case 3 : return User.RoleType.USER_ROLE;
```

Comment:
This should be handled by Enum instead of switch case

Reply:
changed the logic. Added a new function in `RoleType` as `fromValue` which returns the usertype depending upon the value. Else It will return null. Which is handelled by this `getCurrentConversationAssignee() == User.RoleType.BOT ` check

Reply:
Added in [ace8fb1](https://github.com/Kommunicate-io/Kommunicate-Android-Chat-SDK/pull/551/commits/ace8fb1c1b406d3776cfab6178932cc5f973f1b9)